lib/HLSDownloader.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Created by AsherS on 8/20/15.
+ */
+
+/**
+ * Created by AsherS on 8/11/15.
+ */
+
+var Q = require('q');
+var _ = require('underscore');
+var path = require('path');
+var url = require('url');
+var downloadUtils = require('./DownloadUtils');
+var fileUtils = require('./FileUtils');
+var hlsUtils = require('./HLSUtils');
+
+var HLSDownloader = (function () {
+
+    function HLSDownloader(m3u8Url, destPath) {
+        this.streamUrl = m3u8Url;
+        this.destPath = destPath;
+    }
+
+    var downloadIndefinitely = function (url, dest) {
+        downloadPlaylist(url, dest)
+            .then(function(targetDuration){
+                console.log('results: ' + targetDuration);
+            })
+            .catch(function(err){
+                console.log('error ' + err);
+            })
+            .done();
+
+        setTimeout(function () {
+            console.log('-------------------');
+            downloadIndefinitely(url, dest);
+        }, 10000);  //todo, configuration?
+    };
+
+    var downloadPlaylist = function (m3u8Url, dest) {
+
+        var parsedUrl = url.parse(m3u8Url);
+        var baseUrl = parsedUrl.href.substring(0, parsedUrl.href.lastIndexOf('/'));
+        var playlistName = path.basename(parsedUrl.pathname);
```

Comment:
here i assume that i have the complete url to the m3u8 file, hence, i already have the playlist's name